libutils/file_lib.h

```diff
@@ -268,4 +268,36 @@ bool ExclusiveFileLockCheck(FileLock *lock)
 bool ExclusiveFileLockCheck(FileLock *lock);
 #endif  /* __MINGW32__ */
 
+/**
+ * @brief Callback function prototype for PathWalk().
+ * @param dirpath Path to the current directory.
+ * @param dirnames List of subdirectories in dirpath (excluding '.' and '..').
+ * @param filenames List of non-directories in dirpath.
+ * @param data Arbitrary data passed to PathWalk.
+ * @note If you don't want to continue down a path, you can simply remove the
+ *       respective subdirectory from the dirnames sequence, or set it to NULL.
```

Comment:
Let's also mention this can be used to reorder the `dirnames` sequence which otherwise ensures a depth-first walk.
(btw, using `#dirnames` will highlight the string in most editors and would work as a link in generated docs; similarly `%NULL`, for example, can be used for special values)